### DISCUSSION TITLE
Add Mastodon support

### DISCUSSION THREAD

**Discussion by git-sid** [42wim/matterbridge#1724] [New bridges suggestions]
**Is your feature request related to a problem? Please describe.**
Matterbridge doesn't support Mastodon.

**Describe the solution you'd like**
Have Mastodon as a supported messaging channel.

**Describe alternatives you've considered**
Letting the community add the Mastodon support. However, this seems rather difficult without documentation, see #657

**Comment by poVoq** [comment]
Relevant ActivityPub Go library:
https://github.com/go-fed/activity

But IMHO the protocol doesn't lend itself very well to real-time chat. Might be still cool though.

**Comment by Helcaraxan** [comment]
Late to the party but am doing some long overdue issue management.

I'd agree with @poVoq on Mastodon (despite it being very cool) not being well-suited to Matterbridge.

@git-sid would you be able to elaborate on how you would see this integration working in relationship to the other real-time chat protocols?

**Comment by TomTheDragon** [comment]
ActivityPub support would be a great addition to Matterbridge, especially now that many ActivityPub using systems like Mastodon are gaining a lot of users.
It is true that bridging for example a Mastodon account to any realtime chat group, is maybe not the best idea. However on Telegram for example there are "a lot" of channels which people use to post about news, updates and make other kind of announcements. Having a simple way to get this kind of posts over to Mastodon, would be really useful. Especially since there are no other solutions for this kind of problem right now.

**Reply by fauno** [reply]
This is our use case as well!

**Comment by poVoq** [comment]
You can just use the "share to" function with any Mastodon app though, or configure a RSS to Mastodon post bot.

**Reply by TomTheDragon** [reply]
I was talking about the other way around. Like if you want to automate a channel on Telegram, also posting over to a Mastodon account.
